task/pytest/0.2/pytest.yaml

```diff
@@ -0,0 +1,77 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: pytest
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/categories: Testing
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: python, pytest
+    tekton.dev/displayName: pytest
+    tekton.dev/platforms: "linux/amd64,linux/s390x,linux/ppc64le"
+spec:
+  description: >-
+    This task will run pytest on the provided input.
+
+  workspaces:
+    - name: source
+    - description: The workspace consisting of the custom pip settings.
+      name: pip-conf
+      optional: true
+  params:
+    - name: PYTHON
+      description: The used Python version, more precisely the tag for the Python image
+      type: string
+      default: "latest"
+    - name: ARGS
+      description: The additional arguments to be used with pytest
+      type: string
+      default: ""
+    - name: SOURCE_PATH
+      description: The path to the source code
+      default: "."
+    - name: REQUIREMENTS_FILE
+      description: The name of the requirements file inside the source location, with fallback to the requirements file in the root location
+      default: "requirements.txt"
+    - name: PIP_CONF_FILE
+      description: The name of the custom pip config file.
+      default: "pip.conf"
+  steps:
+    - name: unit-test
+      image: docker.io/python:$(inputs.params.PYTHON)
+      workingDir: $(workspaces.source.path)
+      env:
+        - name: PARAM_PIP_CONF_FILE
+          value: $(params.PIP_CONF_FILE)
+        - name: WORKSPACE_PIP_CONF_BOUND
+          value: $(workspaces.pip-conf.bound)
+        - name: WORKSPACE_PIP_CONF_PATH
+          value: $(workspaces.pip-conf.path)
+      script: |
+        export PATH=$PATH:$HOME/.local/bin
+
+        if [ "${WORKSPACE_PIP_CONF_BOUND}" = "true" ] ; then
+          export PIP_CONFIG_FILE="${WORKSPACE_PIP_CONF_PATH}/${PARAM_PIP_CONF_FILE}"
+        fi
+
+        if [ -e "$(inputs.params.SOURCE_PATH)"/"$(inputs.params.REQUIREMENTS_FILE)" ];
```

Comment:
@GijsvanDulmen I think `$(input.params.SOURCE_PATH)` should be as `$(params.SOURCE_PATH)`
Applicable for all the reference in the script
@piyush-garg please confirm

Reply:
That would be different compared to the 0.1 version right?

Reply:
yaa params moved out of input long back

Reply:
@jkandasa @piyush-garg Converted to $(params) ... it does however still works with $(input.params) .. not sure if that is intended though :-)